### PR TITLE
Introduce :include_struct_field option in deriving for Msgpax.Packer

### PR DESCRIPTION
**Pull request by lexmag**

### CODE REVIEW

File: lib/msgpax/packer.ex [hunk 1]
```diff
@@ -236,21 +236,25 @@ defimpl Msgpax.Packer, for: Msgpax.Ext do
...
+  defmacro __deriving__(module, struct, options) do
+    deriving(module, struct, options)
   end
 
-  def deriving(module, opts) do
-    fields = opts[:fields]
+  def deriving(module, struct, options) do
+    keys = struct |> Map.from_struct() |> Map.keys()
+    fields = Keyword.get(options, :fields, keys)
+    include_struct_field? = Keyword.get(options, :include_struct_field, :__struct__ in fields)
```
Comment (lexmag) [inline lib/msgpax/packer.ex]: Maybe use `take_struct_field` instead?

File: lib/msgpax/packer.ex [hunk 2]
```diff
@@ -236,21 +236,25 @@ defimpl Msgpax.Packer, for: Msgpax.Ext do
...
+    fields = List.delete(fields, :__struct__)
     extractor =
       cond do
-        fields && :__struct__ in fields ->
-          fields = List.delete(fields, :__struct__)
+        fields == keys and include_struct_field? ->
+          quote(do: Map.from_struct(struct) |> Map.put("__struct__", unquote(module)))
+        fields == keys ->
+          quote(do: Map.from_struct(struct))
+        include_struct_field? ->
```
Comment (whatyouhide) [inline lib/msgpax/packer.ex]: Can't we kill the `fields == keys` clauses and let `Map.take/2` perform basically the same work as `Map.from_struct/1` for the cases when they are the same?

File: lib/msgpax/packer.ex (hunk 1 above)
Comment (whatyouhide) [inline lib/msgpax/packer.ex]: Personally I really like `include_struct_field` :)

File: lib/msgpax/packer.ex (hunk 2 above)
Comment (lexmag) [inline lib/msgpax/packer.ex]: We can but `Map.from_struct/1` is not quite the same as `Map.take/2`, given that `fields == keys` happens in compile time it's probably fine to assist with having less work in runtime. :man_shrugging:

Comment (whatyouhide) [inline lib/msgpax/packer.ex]: Meh, maybe yeah.

**Pull request merged**